File: medium/cpp/1753.cpp
// https://leetcode.com/problems/maximum-score-from-removing-stones/description/

class Solution
{
public:
    int maximumScore(int a, int b, int c)
    {
        std::priority_queue<int> heap;
        heap.push(a);
        heap.push(b);
        heap.push(c);

        auto res = 0;

        while (std::size(heap) > 1)
        {
            const auto x = heap.top();
            heap.pop();

            const auto y = heap.top();
            heap.pop();

            res++;

            if (x > 1)
            {
                heap.push(x - 1);
            }

            if (y > 1)
            {
                heap.push(y - 1);
            }
        }

        return res;
    }
};

File: medium/python/1753.py
// https://leetcode.com/problems/maximum-score-from-removing-stones/description/

class Solution:
    def maximumScore(self, a: int, b: int, c: int) -> int:
        heap = [-a, -b, -c]
        heapify([a, b, c])
        res = 0

        while len(heap) > 1:
            x = -heappop(heap)
            y = -heappop(heap)
            res += 1
            x -= 1
            y -= 1

            if x > 0:
                heappush(heap, -x)
            if y > 0:
                heappush(heap, -y)

        return res
        